### PR TITLE
Function update

### PR DESCRIPTION
**Pull request by Tess-LaCoil**
**Update to a collection of functions:**
assign_data 
- Now takes either dedicated named vectors or a dataset with appropriately named columns
- Throws errors when the data does not match the required form
- Throws an error when the requested model does not exist
- Checks sizes for vectors in the created data list
- Checks for existence of elements of created data list

run
- Now checks if the chosen model exists in the package

**Selection of new functions:**
extract_estimates
- New function that takes in a model name, matching fit, and returns a list of data frames corresponding to measurement, individual, population (if multi-individual), and error estimated parameters.
- Error handling to address Issue #34
- Error structure: provided fit has to exist and must match the requested model parameter names, and the model must exist.

plot_de_pieces and ggplot_de_pieces
- First is a wrapper for the second that does some data wrangling like find the start and end values of $Y$ for each individual.
- New function that produces a figure of $Y$ against $dY/dt$ based on the chosen model and individual parameters.
- Fits a piece to each individual from $Y_i(0)$ to $Y_i(final)$, parameterised by $\theta_i$
- Has some limited customisability.

model_des
- Function that returns the differential equation as a function for each of the four included models.
- Checks if the model is available in the package.

model_names
- Function that returns a vector containing the names of available models.

model_pars
- Function that returns a list of character vectors for each available model
- Character vectors give parameter names for each level in the model
- Checks if the model is available in the package.

**Test coverage**
New test scripts for assign_data, extract_estimates, and plot_de_pieces.

### CODE REVIEW

**Comment by Tess-LaCoil** [comment]
Tests failed because of the load order of dplyr and stats producing warnings, see lines 70, 71 in the macos checks for an example.

**Comment by Tess-LaCoil** [comment]
I am testing whether removing stats from the imports and using importFrom for the quantile function in the extract_estimates code works to fix this.

File: R/hmde_assign_data.R [hunk 1]
```diff
@@ -1,30 +1,89 @@
...
 #'
 #' @return updated named list with your data assigned to Stan model parameters
 #' @export
 
-hmde_assign_data <- function(model_template, ...){
-  # Grab user expressions
-  user_code <- rlang::enquos(..., .check_assign = TRUE)
+hmde_assign_data <- function(model_template, data = NULL, step_size = NULL, ...){
+  if(!model_template$model %in% hmde_model_name()){
+    stop("Model name not recognised. Run hmde_model_name() to see available models.")
```
Comment (dfalster) [inline R/hmde_assign_data.R:13]: nice work adding all these checks. I didn't look thoroughly, but do you have tests associated with this? Can be good to add a test that checks it fails as intended, e.g.

```
expect_error(hmde_assign_data("Not a model"))
```

**Review by dfalster — APPROVED** [review]
Great work here Tess!

I agree with removing stats from imports, much better to just call in specific functions than entire package.

File: R/hmde_assign_data.R (hunk 1 above)
Comment (Tess-LaCoil) [reply R/hmde_assign_data.R:13]: I went back and checked, had a couple I forgot to add.

**Pull request merged**